drivers/esp-hosted/esp_hosted_proto.h

```diff
@@ -0,0 +1,1849 @@
+/* Generated by the protocol buffer compiler.  DO NOT EDIT! */
```

Comment:
What's the cost/complexity tradeoff of having these files be generated as part of the build process, rather than committing them to the repo?

This would avoid potential bugs where the .proto file and the .c/.h file fall out of sync, and also avoids committing ~6000 lines of auto-generated boilerplate to the source repo.

Reply:
It's possible but it will require installing the protobuf compiler first though, I'm not sure how to add that as a dependency in the extmod Makefile fragment, but maybe @dpgeorge can help with that. Other than that, it's very helpful to be able to see the functions, enums and structs (at least it was when writing the driver).

Reply:
Thanks for getting back to me, @iabdalkader .

If the build system generates this file then they'll still be there to look at, they're just in the build directory instead of checked into git. If you use an IDE-based system for completion (like a VSCode language server) then it will find them and give you autocomplete, type hints, etc. in the same way.

I can take a look at adding the code generation step in the makefile and push a branch with this for consideration.

Reply:
Yes sure, it makes sense to generate them. Note you will also need to add protobuf compiler to the auto-build scripts.

Reply:
If we can auto-generate them as part of the build, that would be a good option (considering how large they are).

Reply:
We just need a rule/dependency on the files, but it has to be in the extmod Make fragment (and protobuf compiler needs to be added to autobuild deps).

Reply:
@iabdalkader @dpgeorge PTAL at:
https://github.com/projectgus/micropython/commit/7c4b10165e5d653226acf8eddad2107e02c4c19a

This builds, but I don't have a board to test it on.

If it looks OK then suggest squashing it into the existing commit in this branch.

Reply:
Looks great, thank you!